rmc/utils/mpc.py

```diff
@@ -517,44 +504,191 @@ def _get_min_aic_glm(
+    # TODO: Make this show/save a table of AICs and model formulas instead of printing every one
 
 
 def calculate_fitted_scores(
     ht: hl.Table,
-    freeze: int = CURRENT_FREEZE,
+    temp_label: str,
+    overwrite_temp: bool = False,
+    fold: int = None,
     interaction_char: str = ":",
     intercept_str: str = "Intercept",
+    freeze: int = CURRENT_FREEZE,
 ) -> hl.Table:
     """
-    Use model and relationship determined in `run_regressions` to calculate fitted scores for input Table.
+    Use MPC model chosen in `run_regressions` to calculate fitted scores for input Table.
 
     .. note::
-        - Assumes input Table has missense observed/expected (`oe`) and `polyphen` annotations.
-        - Function will remove any rows with undefined annotations from input Table.
+        - Function will remove any rows with undefined MPC feature annotations from input Table.
+        - Input Table is assumed to be keyed by ['locus', 'alleles'].
+        - If the model uses transcript-specific features (e.g. RMC O/E) and there are variants
+            in the input HT in multiple transcripts, only the most severe fitted score
+            (and the corresponding annotations it was calculated from) per variant
+            will be retained.
 
     :param hl.Table ht: Input Table with variants to be annotated.
-    :param int freeze: RMC data freeze number. Default is CURRENT_FREEZE.
+    :param str temp_label: Suffix to add to temporary data paths to avoid conflicting names for different models.
+    :param bool overwrite_temp: Whether to overwrite intermediate temporary data if it already exists.
+        If False, will read existing intermediate temporary data rather than overwriting.
+        Default is False.
+    :param int fold: Fold number in training set to select training transcripts from.
+        If not None, the Table is generated from variants in only training transcripts
+        from the specified fold of the overall training set. If None, the Table is generated from
+        variants in all training transcripts. Default is None.
     :param str interaction_char: Character representing interactions in MPC model. Must be one of "*", ":".
         Default is ":".
     :param str intercept_str: Name of intercept variable in MPC model pickle. Default is "Intercept".
-    :return: Table filtered to defined MPC model annotations and annotated with fitted score.
+    :param int freeze: RMC data freeze number. Default is CURRENT_FREEZE.
+    :return: Table annotated with MPC model features and fitted scores, filtered to variants with defined annotations.
     """
     assert interaction_char in {"*", ":"}, "interaction_char must be one of '*' or ':'!"
 
     logger.info("Extracting MPC model relationships from pickle...")
-    model_path = mpc_model_pkl_path(freeze)
-    with hl.hadoop_open(model_path, "rb") as p:
+    with hl.hadoop_open(mpc_model_pkl_path(fold=fold, freeze=freeze), "rb") as p:
```

Comment:
Should there be a check here for the saved model pickle and raise exception if not? Possibly because this function is also called in `annotate_mpc` which could be run independently?

Reply:
Yes, great point